### PR TITLE
Add sub-pixel accuracy to 2D alignment

### PR DESCRIPTION
**Pull request by to266**
resubmission of https://github.com/hyperspy/hyperspy/pull/1236 in order to just quicken things up.

With a test and supports both cross- and phase-correlations.

### CODE REVIEW

**Comment by dnjohnstone** [comment]
Had a quick play with this and it seems ok - although it needs merging up with the current RELEASE_next_minor

File: hyperspy/tests/signal/test_2D_tools.py [hunk 1]
```diff
@@ -58,6 +58,12 @@ def test_align_subpix(self):
         # Compare by broadcasting
         np.testing.assert_allclose(s.data[4], s.data[0], rtol=1)
 
+    def test_estimate_subpix(self):
```
Comment (francisco-dlp) [inline hyperspy/tests/signal/test_2D_tools.py]: The test for the case when ``normalize_corr`` is ``True`` is missing.

**Comment by francisco-dlp** [comment]
Other than the comment above, it looks good to me too.

**Comment by dnjohnstone** [comment]
Could we get this in to v1.2 please? Looks basically done!

**Comment by francisco-dlp** [comment]
It needs a test for phase correlation. Anybody is welcome to contribute it.

**Comment by to266** [comment]
I think when I tried to add the missing test I was not happy with the result - namely even with fine sub-sampling the results were not that much better and not necessarily converged towards the true (known) solution. 

In the end I attempted to debug what the heck was happening, but could not find fault in the actual process. I think it does the correct thing and without mistakes, it's just that the result is not what we'd expect

**Comment by francisco-dlp** [comment]
What I meant is that [this line](https://coveralls.io/builds/9686196/source?filename=hyperspy%2F_signals%2Fsignal2d.py#L101) is not tested.

**Comment by francisco-dlp** [comment]
LGTM, but it requires merging Rnm in.

**Pull request merged**